controllers/operandconfig.go

```diff
@@ -241,29 +250,30 @@ func (r *CommonServiceReconciler) updateOperandConfig(newConfigs []interface{})
 			}
 			if rules != nil && rules.(map[string]interface{})["spec"] != nil && rules.(map[string]interface{})["spec"].(map[string]interface{})[cr] != nil {
 				ruleForCR := rules.(map[string]interface{})["spec"].(map[string]interface{})[cr].(map[string]interface{})
-				opService.(map[string]interface{})["spec"].(map[string]interface{})[cr] = mergeCRsIntoOperandConfig(spec.(map[string]interface{}), newConfigForCR, ruleForCR, overwrite)
+				opService.(map[string]interface{})["spec"].(map[string]interface{})[cr], isChangedWithRules = mergeCRsIntoOperandConfig(spec.(map[string]interface{}), newConfigForCR, ruleForCR, overwrite)
 			} else {
 				if overwrite {
-					opService.(map[string]interface{})["spec"].(map[string]interface{})[cr] = mergeCRsIntoOperandConfigWithDefaultRules(spec.(map[string]interface{}), newConfigForCR)
+					opService.(map[string]interface{})["spec"].(map[string]interface{})[cr], isChangedWithDefultRules = mergeCRsIntoOperandConfigWithDefaultRules(spec.(map[string]interface{}), newConfigForCR)
 				}
 			}
+			isChanged = isChanged || isChangedWithRules || isChangedWithDefultRules
 		}
 	}
 
 	// Checking all the common service CRs to get the minimal size
-	opconServices, err = r.getMinimalSizes(opconServices, ruleSlice)
+	opconServices, _, err = r.getMinimalSizes(opconServices, ruleSlice)
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	opcon.Object["spec"].(map[string]interface{})["services"] = opconServices
 
 	if err := r.Update(ctx, opcon); err != nil {
 		klog.Errorf("failed to update OperandConfig %s: %v", opconKey.String(), err)
-		return err
+		return false, err
 	}
 
```

Comment:
I believe we can add the event here instead of in the reconcile functions.

Reply:
The `r.Recorder.Event` will require instance of  `apiv3.CommonService`.

If we add event in `operandconfig.go`, should we pass the instance as the parameter into the `updateOperandConfig` function